tests/framework/unit_tests/TSA/testLinearRegression.py

```diff
@@ -0,0 +1,299 @@
+        pres = checkSame('',first[i],second[i],update=False)
+      if not pres:
+        print('checking array',comment,'|','entry "{}" does not match: {} != {}'.format(i,first[i],second[i]))
+        res = False
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      results["fail"] += 1
+  return res
+
+def checkNone(comment, entry, update=True):
+  """
+    Checks if entry is None.
+    @ In, comment, string, a comment printed out if it fails
+    @ In, entry, object, to test if against None
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if None
+  """
+  res = entry is None
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking answer",comment,'|','"{}" is not None!'.format(entry))
+      results["fail"] += 1
+
+def checkFails(comment, errstr, function, update=True, args=None, kwargs=None):
+  """
+    Checks if expected error occurs
+    @ In, comment, string, a comment printed out if it fails
+    @ In, errstr, str, expected fail message
+    @ In, function, method, method to run to test for failure
+    @ In, update, bool, optional, if False then don't update results counter
+    @ In, args, list, arguments to pass to function
+    @ In, kwargs, dict, keyword arguments to pass to function
+    @ Out, res, bool, True if failed as expected
+  """
+  print('Error testing ...')
+  if args is None:
+    args = []
+  if kwargs is None:
+    kwargs = {}
+  try:
+    function(*args,**kwargs)
+    res = False
+    msg = 'Function call did not error!'
+  except Exception as e:
+    res = checkSame('',e.args[0],errstr,update=False)
+    if not res:
+      msg = 'Unexpected error message.  \n    Received: "{}"\n    Expected: "{}"'.format(e.args[0],errstr)
+  if update:
+    if res:
+      results["pass"] += 1
+      print(' ... end Error testing (PASSED)')
+    else:
+      print("checking error",comment,'|',msg)
+      results["fail"] += 1
+      print(' ... end Error testing (FAILED)')
+  print('')
+  return res
+
+
+######################################
+#            CONSTRUCTION            #
+######################################
+def createRegressionXML(targets, degree: int):
+  xml = xmlUtils.newNode('Regression', attrib={'target':','.join(targets)})
+  xml.append(xmlUtils.newNode('degree', text=f'{degree}'))
+  return xml
+
+def createFromXML(xml):
+  regression = LR.LinearRegression()
+  inputSpec = LR.LinearRegression.getInputSpecification()()
+  inputSpec.parseNode(xml)
+  regression.handleInput(inputSpec)
+  return regression
+
+def createRegression(targets, degree: int):
+  xml = createRegressionXML(targets, degree)
+  regression = createFromXML(xml)
+  return regression
+
+###################
+#  Simple         #
+###################
+# generate signal
+targets = ['A'] #, 'B', 'C']
+pivot = np.linspace(0, 100, 1000)
+N = len(pivot)
+
+signalA = 20 + (2 * pivot) + (10 * (pivot**2)) + np.random.normal(0, 1, 1000)
```

Comment:
is the random noise just to see how it handles a curveball? It may be worth having a strictly analytic case, and then  adding a noise test as an "advanced" test